test/e2e/gateway_attractor_test.go

```diff
@@ -0,0 +1,347 @@
+	trench := &meridiov1alpha1.Trench{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      trenchName,
+			Namespace: namespace,
+		},
+		Spec: meridiov1alpha1.TrenchSpec{
+			IPFamily: "DualStack",
+		},
+	}
+	attractor := &meridiov1alpha1.Attractor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      attractorName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"trench": trenchName,
+			},
+		},
+		Spec: meridiov1alpha1.AttractorSpec{
+			VlanID:         100,
+			VlanInterface:  "eth0",
+			Replicas:       replicas, // replica of lb-fe
+			Gateways:       []string{"gateway-a", "gateway-b"},
+			Vips:           []string{"vip-a", "vip-b"},
+			VlanPrefixIPv4: "169.254.100.0/24",
+			VlanPrefixIPv6: "100:100::/64",
+		},
+	}
+	gateway := &meridiov1alpha1.Gateway{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "gateway-a",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"attractor": attractorName,
+			},
+		},
+		Spec: meridiov1alpha1.GatewaySpec{
+			Address:  "1.2.3.4",
+			Protocol: "bgp",
+			Bgp: meridiov1alpha1.BgpSpec{
+				RemoteASN:  uint32pointer(1234),
+				LocalASN:   uint32pointer(4321),
+				HoldTime:   "30s",
+				RemotePort: uint16pointer(10179),
+				LocalPort:  uint16pointer(10179),
+			},
+		},
+	}
+	configmapName := fmt.Sprintf("%s-%s", common.CMName, trench.ObjectMeta.Name)
+
+	BeforeEach(func() {
+		fw.CleanUpTrenches()
+		fw.CleanUpAttractors()
+		fw.CleanUpGateways()
+	})
+
+	AfterEach(func() {
+		fw.CleanUpTrenches()
+		fw.CleanUpAttractors()
+		fw.CleanUpGateways()
+	})
+
+	Context("When creating a gateway", func() {
+		AfterEach(func() {
+			fw.CleanUpGateways()
+		})
+		Context("without trench & attractor", func() {
+			It("will be created with disengaged status", func() {
+				Expect(fw.CreateResource(gateway.DeepCopy())).To(Succeed())
+
+				By("checking the existence")
+				gw := &meridiov1alpha1.Gateway{}
+				fw.GetResource(client.ObjectKeyFromObject(gateway), gw)
+				Expect(gw).NotTo(BeNil())
+
+				By("checking the status to be disengaged")
+				Eventually(func() meridiov1alpha1.ConfigStatus {
+					fw.GetResource(client.ObjectKeyFromObject(gateway), gw)
+					return gw.Status.Status
+				}).Should(Equal(meridiov1alpha1.Disengaged))
+			})
+		})
+
+		Context("with trench", func() {
+			BeforeEach(func() {
+				Expect(fw.CreateResource(trench.DeepCopy())).To(Succeed())
+			})
+			JustBeforeEach(func() {
+				Expect(fw.CreateResource(gateway.DeepCopy())).To(Succeed())
+			})
+
+			AfterEach(func() {
+				fw.CleanUpTrenches()
+			})
+
+			Context("with attractor expecting to use this gateway", func() {
+				BeforeEach(func() {
+					Expect(fw.CreateResource(attractor.DeepCopy())).To(Succeed())
+				})
+
+				AfterEach(func() {
+					fw.CleanUpAttractors()
+					fw.CleanUpGateways()
+				})
+
+				It("will be created with engaged status", func() {
+					By("checking if the gateway exists")
+					gw := &meridiov1alpha1.Gateway{}
+					fw.GetResource(client.ObjectKeyFromObject(gateway), gw)
```

Comment:
The errors like here are not always verified

Reply:
They are now fixed.